superbench/analyzer/data_diagnosis.py

```diff
@@ -0,0 +1,359 @@
+        ):
+            baseline['rules'] = {'name': 'variance', 'condition': -0.05}
+        return baseline
+
+    def _get_criteria(self, baseline_file):
+        """Get and generate criteria from metrics.
+
+        Use metric with regex in baseline_file to match the metric full name from raw data
+        for each benchmark enabled in baseline file, and then generate criteria and rule for
+        matched metrics.
+
+        Args:
+            baseline_file (str): The path of baseline yaml file
+
+        Returns:
+            bool: return True if successfully get the criteria, otherwise False.
+        """
+        self._sb_baseline = self._read_baseline(baseline_file)
+        if not self._sb_baseline:
+            return False
+        full_baseline = {}
+        try:
+            self._sb_enable_validation_benchmarks = self.__get_validation_enabled_benchmarks()
+            benchmark_rules = self._sb_baseline['superbench']['benchmarks']
+            for benchmark_name in self._sb_enable_validation_benchmarks:
+                single_benchmark_rules = benchmark_rules[benchmark_name]['metrics']
+                # return_code rule should be a default rule for each benchmark in rule file
+                # if all nodes miss the benchmark and no return code, manually add it
+                if benchmark_name not in self._metrics:
+                    self._metrics[benchmark_name] = {benchmark_name + '/return_code'}
+                # get rules and criteria for each metric
+                for metric in self._metrics[benchmark_name]:
+                    # metric full name in baseline
+                    if metric in single_benchmark_rules:
+                        full_baseline[metric] = single_benchmark_rules[metric]
+                        continue
+                    # metric full name not in baseline, use regex to match
+                    for rule_metric in single_benchmark_rules:
+                        if re.search(rule_metric, metric):
+                            full_baseline[metric] = single_benchmark_rules[rule_metric]
+                            full_baseline[metric] = self._set_default_criteria(full_baseline[metric], metric)
+                            break
+        except Exception as e:
+            logger.error('DataDiagnosis: invalid rule file fomat - {}'.format(str(e)))
+            return False
+
+        self._sb_baseline = full_baseline
+        return True
+
+    def hw_issue(self, benchmark_list):
+        """Idendify if the benchmark is classified as hardware issue.
+
+        All benchmarks except models are classified hardware issue.
+
+        Args:
+            benchmark_list (list): list of benchmarks
+
+        Returns:
+            bool: return true if it's hardware issue
+        """
+        if not isinstance(benchmark_list, set):
+            return False
+        for category in benchmark_list:
+            if 'models' not in category:
```

Comment:
It's incorrect because ORT and TensorRT inference is also microbenchmark.

Reply:
only model and inference(ORT, TensorRT) is not hw issue, is it right?

Reply:
We have remove the hw issue